### ISSUE TITLE
Preprocessing of Assembly101 dataset

### ISSUE THREAD

**Issue by kikizan** [Necolizer/ISTA-Net#5]
Hello, thank you so much for the inspiration your work provides.
I'm having some difficulty processing the Assembly101 dataset, could you please update the ./Preprocess/1_generate_pose_data.py file? 
Thanks!

**Comment by Necolizer** [comment]
Can you provide the errors it throws out when running this script? You can specify the problem you encountered

**Comment by Necolizer** [comment]
@kikizan Hi, I am gonna close this issue after this weekend. Any detailed information about your problems is appreciated and can postpone the scheduled closure.

**Comment by kikizan** [comment]
Hello, I encountered a video_id_json read error when using the 1_generate_pose_data.py file to process data. I made some modifications: annotations_ = pd.read_csv(path_to_csv + split_type + ".csv", sep=',', low_memory=False), the program can be trained smoothly. 
However, the loss has always been nan during the training process. I wonder if you know the reason.
[ 2023-11-14 16:50 ] ###***************start training***************###
[ 2023-11-14 16:50 ] adjust learning rate, using warm up, epoch: 5
[ 2023-11-14 20:08 ] training: epoch: 1, loss: nan, top1: 4.59%, lr: 0.010000
[ 2023-11-14 20:08 ] adjust learning rate, using warm up, epoch: 5
[ 2023-11-14 23:04 ] training: epoch: 2, loss: nan, top1: 4.59%, lr: 0.020000
[ 2023-11-14 23:04 ] adjust learning rate, using warm up, epoch: 5
[ 2023-11-15 01:59 ] training: epoch: 3, loss: nan, top1: 4.59%, lr: 0.030000
[ 2023-11-15 01:59 ] adjust learning rate, using warm up, epoch: 5
[ 2023-11-15 04:51 ] training: epoch: 4, loss: nan, top1: 4.59%, lr: 0.040000
[ 2023-11-15 04:51 ] adjust learning rate, using warm up, epoch: 5
[ 2023-11-15 07:42 ] training: epoch: 5, loss: nan, top1: 4.59%, lr: 0.050000
[ 2023-11-15 08:02 ] evaluating: loss: nan, top1: 4.57%, best_acc: 4.57%

**Comment by Necolizer** [comment]
Okay I think I figure out **part of** the problem. 

1. You can delete all the csv files in 'fine-grained-annotations' (you downloaded from [googledrive](https://drive.google.com/drive/folders/1QoT-hIiKUrSHMxYBKHvWpW9Z9aCznJB7?usp=sharing)) except the file 'test_challenge.csv'. Then pls download the csvs in [this repo](https://github.com/assembly-101/assembly101-action-recognition/tree/main/MS-G3D-action-recognition/CSVs), which are '[train.csv](https://github.com/assembly-101/assembly101-action-recognition/blob/main/MS-G3D-action-recognition/CSVs/train.csv)', '[validation.csv](https://github.com/assembly-101/assembly101-action-recognition/blob/main/MS-G3D-action-recognition/CSVs/validation.csv)', and '[actions.csv](https://github.com/assembly-101/assembly101-action-recognition/blob/main/MS-G3D-action-recognition/CSVs/actions.csv)'. Put them in your previous 'fine-grained-annotations' directory.

This time your directory structure looks like:
```
path/to/your/downdload/root
├── fine-grained-annotations
│   ├── test_challenge.csv  (@30fps)   [This file is download from googledrive]
│   ├── actions.csv                    [This file is download from asb101 repo]
│   ├── train.csv           (@60fps)   [This file is download from asb101 repo]
│   └── validation.csv      (@60fps)   [This file is download from asb101 repo]
```

2. The main difference is the new 'train.csv' and 'validation.csv' are 60 fps, and their headers are also changed. So now don't use your modified line: annotations_ = pd.read_csv(path_to_csv + split_type + ".csv", sep=',', low_memory=False). Get it back:
```
annotations_ = pd.read_csv(path_to_csv + split_type + ".csv", header=0, low_memory=False, names=name_list)
```

3. You can re-run the lines in [**Generating Data**](https://github.com/Necolizer/ISTA-Net#34-assembly101) to re-generate your dataset and splits.

4. As for the mentioned nan-loss problem, I am working on replicating it on my machine. Pls tell me whether it still gets nan after you re-generate your splits and restart your training.

**Comment by kikizan** [comment]
Thank you very much for your answer. After re-dividing the dataset, I solved the problem of loss being nan.
But I found another puzzling problem. My reproduction results on the sbu dataset were much worse than the results given in the paper: clean(95.712), noisy(95.496). 5-fold cross validation approach is adopted.

**Comment by Necolizer** [comment]
Glad to hear that the nan-loss problem problem has been solved. 

In my experiments with torch1.9.0+cu111, GeForce RTX 3070 GPUs and CUDA version 11.4, I can get the avg results of SBU reported in my paper (2 out of 5 folds can even reach 100% acc). I have to say it is really hard and time-consuming to tune the hyper-parameters (e.g. window_size, architecture configs, kernel_size, etc.) FOR SBU DATASET to reach a high acc in my exp. It might probably be the variance when reproducing in a different env with a set of specific tuned hyper-parameters, since SBU dataset is quite small. 

Also, the 5-fold cross validation setting makes each testing split extremely small and very susceptible. Even one more sample in a testing split failed to be recognized can make the acc drop by 2%. So your "much worse" results may be just only 1~2 more samples per split that the model fails to classify correctly. This variance problem still holds if you train and evaluate a different model using SBU, unless doing lots of tuning jobs.

BTW, for the other 3 datasets there can't be a high variance of acc, because they have sufficient testing samples.

I will leave this issue open till this weekend. But don't worry. I will reply all the issues and comments under this repo, though I am not working on action recognition now :)
